app/services/courses/creation_service.rb

```diff
@@ -43,9 +49,8 @@ def build_new_course
         course_enrichment.course_length = '4 years'
       end
 
-      AssignSubjectsService.call(course:, subject_ids:)
+      course.valid?(:new) if course.errors.blank?
```

Comment:
Avoid calling `valid?` on a model when you've added custom errors.

Calling `valid?` will clear your custom errors unless they are defined on the model as validations.